File: android/app/src/main/kotlin/com/zaplivre/ui/components/ReactionBar.kt
package com.zaplivre.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Reaction count data class
 */
data class ReactionCount(
    val emoji: String,
    val count: Int,
    val hasReacted: Boolean
)

/**
 * ReactionBar - Displays emoji reactions under a message
 *
 * Shows aggregated reactions with counts and highlights user's reactions.
 */
@Composable
fun ReactionBar(
    reactions: List<ReactionCount>,
    onReactionClick: (String) -> Unit,
    onAddReactionClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    if (reactions.isEmpty()) {
        return
    }

    LazyRow(
        modifier = modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        contentPadding = PaddingValues(horizontal = 8.dp, vertical = 4.dp)
    ) {
        items(reactions) { reaction ->
            ReactionChip(
                emoji = reaction.emoji,
                count = reaction.count,
                isSelected = reaction.hasReacted,
                onClick = { onReactionClick(reaction.emoji) }
            )
        }

        // Add reaction button
        item {
            Surface(
                shape = RoundedCornerShape(12.dp),
                color = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f),
                modifier = Modifier
                    .height(28.dp)
                    .clickable(onClick = onAddReactionClick)
            ) {
                Row(
                    modifier = Modifier.padding(horizontal = 8.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.Add,
                        contentDescription = "Add reaction",
                        modifier = Modifier.size(16.dp),
                        tint = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
            }
        }
    }
}

/**
 * ReactionChip - Individual reaction bubble
 */
@Composable
fun ReactionChip(
    emoji: String,
    count: Int,
    isSelected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Surface(
        shape = RoundedCornerShape(12.dp),
        color = if (isSelected) {
            MaterialTheme.colorScheme.primaryContainer
        } else {
            MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.5f)
        },
        modifier = modifier
            .height(28.dp)
            .clickable(onClick = onClick)
    ) {
        Row(
            modifier = Modifier.padding(horizontal = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            Text(
                text = emoji,
                fontSize = 14.sp
            )
            Text(
                text = count.toString(),
                fontSize = 12.sp,
                color = if (isSelected) {
                    MaterialTheme.colorScheme.onPrimaryContainer
                } else {
                    MaterialTheme.colorScheme.onSurfaceVariant
                }
            )
        }
    }
}

File: ios/ZapLivre/ZapLivre/Components/ReactionBar.swift
//
//  ReactionBar.swift
//  ZapLivre
//
//  Created by ZapLivre Team
//  Copyright © 2026 ZapLivre. All rights reserved.
//

import SwiftUI

/// Reaction count model
struct ReactionCount: Identifiable {
    let id = UUID()
    let emoji: String
    let count: Int
    let hasReacted: Bool
}

/// ReactionBar - Displays emoji reactions under a message
///
/// Shows aggregated reactions with counts and highlights user's reactions.
struct ReactionBar: View {
    let reactions: [ReactionCount]
    let onReactionTap: (String) -> Void
    let onAddReactionTap: () -> Void

    var body: some View {
        if !reactions.isEmpty {
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 6) {
                    ForEach(reactions) { reaction in
                        ReactionChip(
                            emoji: reaction.emoji,
                            count: reaction.count,
                            isSelected: reaction.hasReacted,
                            onTap: { onReactionTap(reaction.emoji) }
                        )
                    }

                    // Add reaction button
                    Button(action: onAddReactionTap) {
                        Image(systemName: "plus")
                            .font(.system(size: 12, weight: .medium))
                            .foregroundColor(.secondary)
                            .frame(height: 24)
                            .padding(.horizontal, 8)
                            .background(Color.secondary.opacity(0.1))
                            .cornerRadius(12)
                    }
                }
                .padding(.horizontal, 8)
                .padding(.vertical, 4)
            }
        }
    }
}

/// ReactionChip - Individual reaction bubble
struct ReactionChip: View {
    let emoji: String
    let count: Int
    let isSelected: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 4) {
                Text(emoji)
                    .font(.system(size: 14))

                Text("\(count)")
                    .font(.system(size: 12, weight: .medium))
                    .foregroundColor(isSelected ? .white : .secondary)
            }
            .padding(.horizontal, 8)
            .frame(height: 24)
            .background(isSelected ? Color.blue : Color.secondary.opacity(0.1))
            .cornerRadius(12)
        }
    }
}

#Preview {
    VStack(spacing: 12) {
        ReactionBar(
            reactions: [
                ReactionCount(emoji: "👍", count: 3, hasReacted: true),
                ReactionCount(emoji: "❤️", count: 2, hasReacted: false),
                ReactionCount(emoji: "😂", count: 1, hasReacted: false)
            ],
            onReactionTap: { emoji in print("Tapped: \(emoji)") },
            onAddReactionTap: { print("Add reaction") }
        )
        .padding()
        .background(Color.gray.opacity(0.1))

        ReactionBar(
            reactions: [],
            onReactionTap: { _ in },
            onAddReactionTap: {}
        )
    }
}
